internal/convert/automation_config.go

```diff
@@ -31,12 +31,10 @@ func FromAutomationConfig(c *opsmngr.AutomationConfig) []*ClusterConfig {
 		}
 
 		newSC.Config = newRSConfig(c, s.ConfigServerReplica)
-		for j, p := range c.Processes {
-			if p.Cluster == s.Name {
+		for _, p := range c.Processes {
+			if p.Cluster == s.Name && p.ProcessType == "mongos" {
 				newSC.Mongos = append(newSC.Mongos, newMongosProcessConfig(p))
 				newSC.addToMongoURI(p)
-				c.Processes = removeProcess(c.Processes, j)
```

Comment:
@gssbzn I believe `removeProcess` is not required but if this was for optimization then I can rewrite this logic.

Reply:
yes, removeProcess was an optimization, processes can be part of a RS, a stand alone or a shared cluster and you can have multiple deployments in a single automation config, removing here was to avoid looking at process that were already linked to one of the above

Reply:
Thanks for the clarification. I have updated the code to removeProcess but that has changed the sequence of processes in MongoURI and mongos object. I hope that is fine.